pkg/controllers/reconciler.go

```diff
@@ -159,6 +167,72 @@ func (r *reconciler) createObjects(ctx context.Context,
 	return result, nil
 }
 
+func imageGetVersion(image string) string {
+	version := ""
+
+	parts := strings.SplitN(filepath.Base(image), ":", 2)
+
+	if len(parts) == 2 {
+		version = parts[1]
+	}
+
+	return version
+}
+
+func imageGetName(image string) string {
+	name := ""
+
+	parts := strings.SplitN(filepath.Base(image), ":", 2)
+
+	if len(parts) == 2 {
+		name = parts[0]
+	}
+
+	return name
+}
+
+func upgradeToImage(image string) string {
+	out := ""
+
+	ver, e1 := version.ParseSemantic(imageGetVersion(image))
+	minVer, e2 := version.ParseSemantic(ImageMinVersion)
+
+	if e1 == nil && e2 == nil && ver.LessThan(minVer) {
+		out = "intel/" + imageGetName(image) + ":" + ImageMinVersion
```

Comment:
nit: if imageGetName returns empty string we'll get 'intel/:<version>' as a result.